projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php

```diff
@@ -36,15 +36,15 @@ public static function is_enabled_jetpack_seo( $site_id = 0 ) {
 		}
 
 		if ( function_exists( 'has_any_blog_stickers' ) ) {
-			// For WPCOM sites
+			// For WPCOM sites.
 			if ( empty( $site_id ) ) {
 				$site_id = get_current_blog_id();
 			}
 
 			return has_any_blog_stickers( array( 'business-plan', 'ecommerce-plan' ), $site_id );
 		}
 
-		// For all Jetpack sites
+		// For all Jetpack sites.
 		return true;
```

Comment:
I believe you may be able to refactor `is_enabled_jetpack_seo` since we don't need to check for a plan anymore. We may even be able to get rid of it entirely alongside the `jetpack_disable_seo_tools` filter, and instead ensure that everything is shortcircuited early when the module is not active or when a conflicting plugin is installed (see `projects/plugins/jetpack/modules/seo-tools.php`).

Reply:
@jeherve Nice catch, from what I tested the plan check is not needed in `is_enabled_jetpack_seo()` so it's been removed in https://github.com/Automattic/jetpack/pull/17309/commits/073a7385b98951936c1f85499fef0550bc0e0075

> We may even be able to get rid of it entirely alongside the jetpack_disable_seo_tools filter...

I'm a bit more hesitant with making this change, as the filter comment mentions:

> Can be used by SEO plugin authors to disable the conflicting output of SEO Tools.

I'm not sure of the likelihood of that actually being used, and don't think it would hurt leaving it in, but I'll defer to your expertise on this.

Reply:
Yeah, we could keep the filter for a bit longer. Maybe we could move it to `projects/plugins/jetpack/modules/seo-tools.php` instead of having `is_enabled_jetpack_seo()`, and shortcircuit things early (e.g. not even including `seo-tools/jetpack-seo.php`) when the filter is in use, or when a conflicting plugin is installed.

Reply:
I see the static method in places like:

https://github.com/Automattic/jetpack/blob/222e0ec1fb9b7984bbcb0d4b759c457961d9bc1b/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php#L795

Which is called on the Calypso side when saving a new front page meta description for example. Should we leave that is for now? I did add the filter check in the `seo-tools.php` file in https://github.com/Automattic/jetpack/pull/17309/commits/3a5da042409614a7b47bc461d63e163441263816  that look ok?

Reply:
Yeah 👍